app/src/main/java/com/spascoding/taskycourse/feature_register_screen/presentation/login/LoginViewModel.kt

```diff
@@ -30,11 +37,15 @@ class LoginViewModel @Inject constructor(
                 )
             }
             is LoginEvent.LoginAction -> {
-                authenticationUseCases.loginUser.invoke(
-                    email = state.email,
-                    password = state.password,
-                ) {
-                    event.onSuccess.invoke(it.userId)
+                GlobalScope.launch(Dispatchers.IO) {
+                    val response = authenticationUseCases.loginUser.invoke(
+                        email = state.email,
+                        password = state.password,
+                    )
+                    if (response.isSuccessful.not()) {
```

Comment:
Do you find this more readable than `!response.isSuccessful`? Why not use the native operators?